plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessor.java

```diff
@@ -144,20 +152,106 @@ public void processResponseAsync(
                 responseListener.onResponse(response);
                 return;
             }
-            rewriteResponseDocuments(response, responseListener);
+
+            // if many to one, run rewriteResponseDocuments
+            if (!oneToOne) {
+                rewriteResponseDocuments(response, responseListener);
+            } else {
+                // if one to one, make one hit search response and run rewriteResponseDocuments
+                GroupedActionListener<SearchResponse> combineResponseListener = getCombineResponseGroupedActionListener(
+                    response,
+                    responseListener,
+                    hits
+                );
+
+                for (SearchHit hit : hits) {
+                    SearchHit[] newHits = new SearchHit[1];
+                    newHits[0] = hit;
+                    SearchResponse oneHitResponse = SearchResponseUtil.replaceHits(newHits, response);
+                    ActionListener<SearchResponse> oneHitListener = getOneHitListener(combineResponseListener);
+                    rewriteResponseDocuments(oneHitResponse, oneHitListener);
+                }
+            }
+
         } catch (Exception e) {
             if (ignoreFailure) {
                 responseListener.onResponse(response);
             } else {
-                responseListener.onFailure(e);
+                responseListener.onFailure(new RuntimeException(e.getMessage()));
```

Comment:
Why encapsulate the original exception to RuntimeException?

Reply:
to avoid 500x error

Reply:
Suggest use OpenSearchStatusException to replace RuntimeException.

Reply:
I don't see 5xx exception, it's for precautions. changed to OpenSearchStatusException similar to https://github.com/opensearch-project/ml-commons/blob/9663053d544b59ef238ae36bf49b774543552df5/plugin/src/main/java/org/opensearch/ml/utils/IndexUtils.java#L134


added in commit [use OpenSearchStatusException in error handling](https://github.com/opensearch-project/ml-commons/pull/2801/commits/0d5c6bf66f87f1819ae4554b55d278ea0fa7c5ec)